### ISSUE TITLE
Name change proposal ofDrawRectRounded -> ofDrawRoundedRectangle

### ISSUE THREAD

**Issue by bakercp** [openframeworks/openFrameworks#4163]
We deprecated `ofRect` and it became `ofDrawRectangle`.  That was super, clear and concise.  In fact almost all of our drawing methods are clear and concise now.  But ... for some reason we deprecated `ofRectRounded` and made it `ofDrawRectRounded`.  

I'd propose that we'd instead replace it with `ofDrawRoundedRectangle`.  It seems that if we are going to do this we might want to consider doing it before 0.9.0 since we deprecated `ofRectRounded` during this release cycle.

**Comment by bakercp** [comment]
Alternatively, I would be interested in overloading `ofDrawRectangle` to take the additional arguments needed to make it draw rounded rectangles.

**Comment by arturoc** [comment]
i wonder if we should just deprecate ofRectRounded completely and not add it back, it's available through ofPath and creating it every frame is really slow, it's been reported a couple of times in the forum by people trying to use it in immediate mode. anything more complex than the basic primitives is better in ofPath were it can be cached

**Comment by bakercp** [comment]
Hm.  Not sure.  If you're changing the values of the rectangle based on some input, it's not going to make much difference is it?  But from a philosophical perspective, limiting the immediate mode drawing functions to the most basic primitives (or cacheable meshes that can be simply transformed) makes sense.

**Comment by kylemcdonald** [comment]
for discoverability via autocomplete i would prefer `ofDrawRectangleRounded`

i'd be curious to hear @ofzach @phoenixperry or other teachers chime in about changing it to ofPath-only.

**Comment by bakercp** [comment]
@kylemcdonald yeah, I agree with that -- I think we have discussed that before somewhere.  I reversed it because it was easier to say, but discoverability would win I think.  Still the larger issue of moving it to ofPath remains ...

**Comment by arturoc** [comment]
if the shape is dynamic you can pull the mesh out of the ofPath and put it on an of3dPrimitive to transform it instead of regenerating a full circle every time. we really need a better API to do that but with immediate mode you can't do it at all.

The rounded rectangle is the only shape (along with ofBeginShape/ofEndShape) in immediate mode that is not drawn directly to the gpu but needs a ofPath to be drawn which is already a sign that it's a really special case.

Even pedagogically i think it makes sense, i usually teach basic drawing, screen coordinates and basic transformations,  using basic shapes like a circle, rectangle. A rectangle with round corners is a really special case. If someone wants to draw a rectangle with rounded corners i would point them to ofPath and explain how it's a special case because of how complex it's to draw (lot's of sines and cosines, and really difficult to cache for the general case) and introduce the idea that in some cases it makes sense to create the geometry first, once and draw that stored geometry every frame as opposed to calculate it every frame, which makes sense for anything that is not a really basic primitive
